airflow/jobs.py

```diff
@@ -2001,6 +2019,12 @@ def _get_dag_run(self, run_date, session=None):
             )
         else:
             run = run[0]
+            if run.state == State.SUCCESS:
+                return run
+
+            if (run.state != State.RUNNING and
```

Comment:
Discussed offline, we are going to remove success as a special case dagrun state here to make this function simpler.